apollo-router/src/router/event/mod.rs

```diff
@@ -29,7 +30,7 @@ use crate::uplink::schema::SchemaState;
 /// Messages that are broadcast across the app.
 pub(crate) enum Event {
     /// The configuration was updated.
-    UpdateConfiguration(Configuration),
+    UpdateConfiguration(Arc<Configuration>),
```

Comment:
Used an `Arc` here only because we also have `Arc<Configuration>` throughout the router. A `Box` would also be fine but 🤷🏻‍♀️